linode/helper/regex_test.go

```diff
@@ -0,0 +1,42 @@
+//go:build unit
+
+package helper_test
+
+import (
+	"testing"
+
+	"github.com/linode/terraform-provider-linode/linode/helper"
+)
+
+func TestCheckRegex(t *testing.T) {
+	pattern := "^[a-zA-Z0-9]([-_.]?[a-zA-Z0-9]+)*[a-zA-Z0-9]$"
```

Comment:
This regex pattern looks correct and should be good to implement in a validator 🚀